eval_mixer.py

```diff
@@ -0,0 +1,218 @@
+            )
+        outputs_s2s = outputs_s2s.cpu().numpy()
+        probs_s2s = probs_s2s.cpu().numpy()
+        outputs_linear = outputs_linear.cpu().numpy()
+        probs_linear = probs_linear.cpu().numpy()
+
+        # Store input in its original string format
+        inputs.extend(batch['occ1'])
+
+        # Store predictions and associated "probabilities"
+        preds_s2s_raw.append(outputs_s2s)
+        probs_s2s_raw.append(probs_s2s)
+        preds_linear_raw.append(outputs_linear)
+        probs_linear_raw.append(probs_linear)
+
+        # Store labels
+        targets = targets.detach().cpu().numpy()
+        labels_raw.append(targets)
+
+        batch_time.update(time.time() - end)
+
+        if batch_idx % 10 == 0 or batch_idx == 1:
+            print(f'Finished prediction for batch {batch_idx} of {len(data_loader)}')
+            print(f'Batch time (data): {batch_time.avg:.2f} ({batch_time_data.avg:.2f}).')
+            print(f'Max. memory allocated/reserved: {torch.cuda.max_memory_allocated() / (1024 ** 3):.2f}/{torch.cuda.max_memory_reserved() / (1024 ** 3):.2f} GB')
+
+        end = time.time()
+
+    preds_s2s_raw = np.concatenate(preds_s2s_raw)
+    probs_s2s_raw = np.concatenate(probs_s2s_raw)
+
+    preds_linear_raw = np.concatenate(preds_linear_raw)
+    probs_linear_raw = np.concatenate(probs_linear_raw)
+
+    labels_raw = np.concatenate(labels_raw)
+
+    preds_s2s = list(map(
+        data_loader.dataset.formatter.clean_pred,
+        preds_s2s_raw,
+    ))
+    preds_linear = np.array([
+        [data_loader.dataset.formatter.lookup_hisco[x] for x in y]
+        for y in preds_linear_raw
+        ])
+    labels = list(map(
+        data_loader.dataset.formatter.clean_pred,
+        labels_raw,
+    ))
+
+    preds = pd.DataFrame({
+        'input': inputs,
+        'label': labels,
+        'pred_s2s': preds_s2s,
+        **{f'pred_linear_{i}': preds_linear[:, i] for i in range(linear_topk)},
+        **{f'prob_s2s_{i}': probs_s2s_raw[:, i] for i in range(probs_s2s_raw.shape[1])},
+        **{f'prob_linear_{i}': probs_linear_raw[:, i] for i in range(linear_topk)},
+    })
+
+    preds.to_csv(fn_out, index=False)
+
+
+def main():
+    args = parse_args()
+
+    # Target-side tokenization
+    formatter = blocky5()
+
+    # Input-side tokenization
+    tokenizer = load_tokenizer(
+        model_domain='Multilingual_CANINE',
+    )
+
+    # Dataset
+    dataset_val = setup_dataset(
+        args=args,
+        formatter=formatter,
+        tokenizer=tokenizer,
+    )
+
+    # Data loader
+    data_loader_val = DataLoader(
+        dataset_val,
+        batch_size=args.batch_size,
+        shuffle=True,
+        num_workers=args.num_workers,
+        pin_memory=args.pin_memory,
+        )
+
+    # Setup model
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+
+    model = Seq2SeqMixerOccCANINE(
+        model_domain='Multilingual_CANINE', # TODO make arg, discuss with Vedel
```

Comment:
We should probably just make a decision here.

Reply:
View on whether going with arg that defaults to "Multilingual_CANINE" or keep it hardcoded as for now? Since this is not part of the package (but rather just an entrypoint script), I think harcoding is fine.

Reply:
I see. That does make sense!